dash/frontend/src/app/core/services/falco.service.ts

```diff
@@ -17,77 +31,77 @@ export class FalcoService {
     private httpClient: HttpClient,
   ) {}
 
-  getFalcoLogs(clusterId: number,
-               limit?: number,
-               page?: number,
-               selectedPriorityLevels?: string [],
-               selectedOrderBy?: string,
-               startDate?: string,
-               endDate?: string,
-               namespace?: string,
-               pod?: string,
-               image?: string
+  getFalcoLogs(clusterId: number, options?: FalcoLogOptions
   ): Observable<IServerResponse<{logCount: number, list: IFalcoLog[]}>> {
-    const params = this.buildParams(clusterId, limit, page, selectedPriorityLevels, selectedOrderBy, startDate, endDate, namespace, pod, image);
+    const params = this.buildParams(clusterId, options);
     return this.httpClient.get('/api/falco', {params});
   }
 
+  getFalcoLogByEventId(eventId: number
+  ): Observable<IServerResponse<IFalcoLog>> {
+    return this.httpClient.get(`/api/falco/${eventId}`);
+  }
+
+  getCountOfFalcoLogsBySignature(clusterId: number, signature: string
+  ): Observable<IServerResponse<IFalcoCount[]>>{
+    const params = this.buildParams(clusterId, {signature} );
+    return this.httpClient.get('/api/falco/count', {params});
+  }
 
   downloadFalcoExport(clusterId: number):
     Observable<IServerResponse<IFalcoCsv>> {
     const params = this.buildParams(clusterId);
     return this.httpClient.get('/api/falco/download', {params});
   }
 
-  buildParams(clusterId: number,
-              limit?: number,
-              page?: number,
-              selectedPriorityLevels?: string [],
-              selectedOrderBy?: string,
-              startDate?: string,
-              endDate?: string,
-              namespace?: string,
-              pod?: string,
-              image?: string
+  buildParams(clusterId: number, options?: FalcoLogOptions
   ): HttpParams {
     let params: HttpParams = new HttpParams();
     params = params.set('cluster-id', clusterId.toString());
 
-    if (limit){
-      params = params.set('limit', String(limit));
+    if (options?.limit){
+      params = params.set('limit', String(options?.limit));
```

Comment:
Lets make sure the limit and page count are numeric.

Reply:
I think you can multiply it by 1 to force it into being a number. Kyle may have another way, too. In SQL normally the limit/page count/sort by are not escaped.